File: app/src/main/java/com/mobarak/todo/AppApplication.java
package com.mobarak.todo;

import android.app.Application;
import android.content.Context;

public class AppApplication extends Application {
    private static AppApplication mInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
    }

    public static AppApplication getInstance() {
        return mInstance;
    }

    public static Context getAppContext() {
        return mInstance.getApplicationContext();
    }
}

File: app/src/main/java/com/mobarak/todo/ui/base/AppApplication.kt
package com.mobarak.todo.ui.base

import android.app.Application
import android.content.Context

class AppApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {
        private lateinit var instance: AppApplication

        val appContext: Context
            get() = instance.applicationContext
    }
}